src/__tests__/screens/navigation.test.tsx

```diff
@@ -23,13 +23,24 @@ describe('Navigation Test Suite', async () => {
 
         // check home page
         await waitFor(() => screen.getByTestId('featured-search-heading'));
-        expect(screen.getByTestId('featured-search-button')).toBeInTheDocument();
+        // get all elements with the "featured-search-button" testid
+        const buttons = screen.getAllByTestId('featured-search-button');
+        // check for the first button
+        expect(buttons[0]).toBeInTheDocument();
+        // open user menu
+        await user.click(screen.getByTestId('menu-button'));
+        // wait for the menu to be displayed
+        await waitFor(() => screen.getByTestId('menu-appbar'));
         // check login
         await user.click(screen.getByText('Login'));
         await waitFor(() => screen.getByTestId('login-heading'));
         // go back to homepage
-        await user.click(screen.getByText('Home'));
+        await user.click(screen.getByTestId('home'));
         await waitFor(() => screen.getByTestId('featured-search-heading'));
+        // open user menu
+        await user.click(screen.getByTestId('menu-button'));
+        // wait for the menu to be displayed
+        await waitFor(() => screen.getByTestId('menu-appbar'));
```

Comment:
We could consider abstracting this bit of code since it is used so many times. We could create a util file that houses common actions like this. This can be turned into a new ticket, not a blocker.